src/lazy/expanded/macro_table.rs

```diff
@@ -257,65 +280,137 @@ impl MacroTable {
-                        ParameterCardinality::ZeroOrMore,
+                        ParameterCardinality::ExactlyOne,
                         RestSyntaxPolicy::NotAllowed,
                     ),
                     Parameter::new(
-                        "value_to_annotate",
+                        "exponent",
                         ParameterEncoding::Tagged,
                         ParameterCardinality::ExactlyOne,
                         RestSyntaxPolicy::NotAllowed,
                     ),
                 ])
                 .unwrap(),
-                MacroKind::Annotate,
-                ExpansionAnalysis {
-                    could_produce_system_value: true,
-                    must_produce_exactly_one_value: true,
-                    can_be_lazily_evaluated_at_top_level: false,
-                    expansion_singleton: None,
-                },
+                MacroKind::ToDo,
+                ExpansionAnalysis::single_application_value(IonType::Decimal),
+            )),
+            Arc::new(Macro::named(
+                "make_timestamp",
+                MacroSignature::new(vec![
+                    Parameter::new(
+                        "year",
+                        ParameterEncoding::Tagged,
+                        ParameterCardinality::ExactlyOne,
+                        RestSyntaxPolicy::NotAllowed,
+                    ),
+                    Parameter::new(
+                        "month",
+                        ParameterEncoding::Tagged,
+                        ParameterCardinality::ZeroOrOne,
+                        RestSyntaxPolicy::NotAllowed,
+                    ),
+                    Parameter::new(
+                        "day",
+                        ParameterEncoding::Tagged,
+                        ParameterCardinality::ZeroOrOne,
+                        RestSyntaxPolicy::NotAllowed,
+                    ),
+                    Parameter::new(
+                        "hour",
+                        ParameterEncoding::Tagged,
+                        ParameterCardinality::ZeroOrOne,
+                        RestSyntaxPolicy::NotAllowed,
+                    ),
+                    Parameter::new(
+                        "minute",
+                        ParameterEncoding::Tagged,
+                        ParameterCardinality::ZeroOrOne,
+                        RestSyntaxPolicy::NotAllowed,
+                    ),
+                    Parameter::new(
+                        "second",
+                        ParameterEncoding::Tagged,
+                        ParameterCardinality::ZeroOrOne,
+                        RestSyntaxPolicy::NotAllowed,
+                    ),
+                    Parameter::new(
+                        "offset_minutes",
+                        ParameterEncoding::Tagged,
+                        ParameterCardinality::ZeroOrOne,
+                        RestSyntaxPolicy::NotAllowed,
+                    ),
```

Comment:
This stretch makes me wonder if named `Parameter` constructors would also be helpful. Is the required cardinality too high? I would think that at least `tagged`, `tagged_optional`, and `tagged_rest` would be helpful (roughly covering `x`, `x?`, and `x*`).

This also makes me think that the system macro table might be a good place to point people at for usage examples of parameters, we run a gamut here. Not as much as we used to with tagged encodings for the system macros, but still there's a fair bit of diversity of use here.